### ISSUE TITLE
Stack doesn't work on Android

### ISSUE THREAD

**Issue by billyohgren** [wix/react-native-navigation#2685]
### Issue Description

I'm trying to run an Android app with a StackViewController but all I get is a white screen with a topBar. Running this in iOS works though.

### Steps to Reproduce / Code Snippets / Screenshots
This won't work on Android (but works in iOS):
```Navigation.setRoot(
stack: {
    name: 'Test',
    children: [{
      component: {
        name: 'testScreen',
        passProps: {
        },
      },
    },
    ],
  },
)
```

This will display my testScreen on Android and iOS (but won't let me push because it's not a stack):
```
component: {
        name: 'swadd.testScreen',
        passProps: {
        },
     },
```
---
### Environment
* React Native Navigation version: 2.0.2119
* React Native version: 0.50.3
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Samsung S8. 7.0. Debug.

**Comment by billyohgren** [comment]
After countless error of debugging I solved this by changing:
```getView().addView(enteringView);```
to
```getView().addView(enteringView, LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT);```
in StackController.java in both ```public void push(ViewController child, final Promise promise)``` and ```public void animatePush(final ViewController child, final Promise promise)```.

Since I'm no Android developer it would be nice to get some feedback on this fix, I'll create a PR and submit later today.

**Comment by RubenSandwich** [comment]
This is also required for the screen to resize back to full screen after a keyboard dismisses.

**Comment by guyca** [comment]
Hey @billyohgren 

The following works in the playground app
```js
    Navigation.setRoot({
      stack: {
        children: [
          {
            component: {
              name: 'navigation.playground.WelcomeScreen'
            }
          }
        ]
      }
    });
```

Can you please paste the complete api call?

**Comment by viotti** [comment]
Similar problem here, but with `bottomTabs`. I even posted a question on Stack Overflow.

https://stackoverflow.com/q/48724828/2619107

I get a white screen with the bottom bar on Android.

**Comment by dviluk** [comment]
same problem here, with stack based. I get a white screen and the navbar.

**Comment by JohnSmith0602** [comment]
Same issue here with v2.0.2121, ios works fine, while android renders white screen and the navbar.

The api call is its simplest form:

javascript
```
Navigation.events().onAppLaunched(() => {
  Navigation.setRoot({
    stack: {
      children: [{
        component: {
          name: 'Discover'
        }
      }]
  })
})
```

**Comment by JohnSmith0602** [comment]
The only additional info I can think of is the configuration that I follow [this link](https://github.com/wix/react-native-navigation/issues/2354#issuecomment-354930325) to get the android build run successfully, hope it helps.

**Comment by billyohgren** [comment]
@guyca Thank you for replying! Using your snippet in my code (but replacing screen name ofc) does not work, I just end up with a white screen. This https://github.com/wix/react-native-navigation/pull/2686 solved it for me. Could it be that we're using different versions of the Android SDK?

@JohnSmith0602 so you're saying that you solved the white screen by following instructions in that link, or that the android build runs (at all?)

**Comment by JohnSmith0602** [comment]
@billyohgren No, I mean I am facing the same issue using the configuration in the link.

**Comment by viotti** [comment]
I am also using the configuration posted by fvicente on https://github.com/wix/react-native-navigation/issues/2354#issuecomment-354930325.

**Comment by viotti** [comment]
And I solved the `bottomTabs` problem with the same technique used by @billyohgren (thanks, BTW).

I changed `getView().addView(getCurrentView());` to `getView().addView(getCurrentView(), MATCH_PARENT, MATCH_PARENT);` on method `selectTabAtIndex` of the `com.reactnativenavigation.viewcontrollers.BottomTabsController` class. The `diff` to version RNN 2.0.2125 is located at https://stackoverflow.com/a/48756995/2619107.

Now a tab change causes the component to mount.

I can make another PR or maybe @billyohgren could add an extra line to https://github.com/wix/react-native-navigation/pull/2686.
